pkg/clusterd/disk.go

```diff
@@ -113,6 +118,38 @@ func DiscoverDevices(executor exec.Executor) ([]*sys.LocalDisk, error) {
 	return disks, nil
 }
 
+func deviceMatchWithFilter(device string, filter, metaDevice string) bool {
+	if filter == listAllDevices || device == metaDevice {
+		return true
+	}
+
+	// We need to allow dm- devices (Kubernetes mountpoints) because dm- devices name are used as
+	// name pattern for meta device
+	pattern, _ := regexp.Compile("dm")
```

Comment:
I made that change but the linter is complain on ineffectual assignment to err

```
pkg/clusterd/disk.go:133:11: ineffectual assignment to err (ineffassign)
	pattern, err = regexp.Compile("dm")
```

seems the issue is caused due always the error will be nil, so I should ignore that error by using (_) , what do you think?

Reply:
How about adding the error check?
```
if err != nil {
  return false
}
```

Reply:
@travisn @Javlopez  Let's make the `dm` regex a var at the top of the file using `regexp.MustCompile` instead of the `regexp.Compile`.

This simple regex should always compile, should it ever not compile Golang regex package has broken compatibility in some way.